### ISSUE TITLE
Cost allocation based on label/tags dos not work

### ISSUE THREAD

**Issue by serakis** [opencost/opencost#1186]
In our cluster we have kubernetes labels and AWS resources tags (like s3). The goal is to get an overview of both k8s cost an AWS cost for a certain product using labels and tags but also review deeper in AWS and k8s which parts that cost money for a certain product.
 
We followed the guide https://guide.kubecost.com/hc/en-us/articles/4407596810519-AWS-Out-of-Cluster- and set in General settings **Product Label**=product (same as k8s label) and added labels/tags as shown below:

Kubernetes deployment: product=name1 
AWS resources: kubernetes_label_product=name1

This works for k8s but not for AWS at all. No external cost was seen but we could search for the AWS label in Assets and find that the AWS resources was tagged.

Since the guide did not work, we started to explore a bit and added tags and labels that did not follow the guide mentioned above.

**Scenario 1: Same AWS tag and K8s label** 
General settings Product Label=xx_product
Kubernetes deployment: xx_product=name1 
AWS resources: xx_product=name1
```
labelMappingConfigs:
          enabled: true
          product_label: xx_product
          product_external_label: xx_product
```
This configuration worked for Cost allocation setting **Aggregate by** = Product. All cost for k8s and external cost was updated to the overview. However when you want to inspect external costs (by pressing the three dots -> Inspect External Costs) the tag changed from expected **xx_product** to **kubernetes_product** adding an extra **kubernetes_** to the tag and removing **xx_**.

**Scenario 2: Skip only _label in AWS resource** 
General settings Product Label=xx_product
Kubernetes deployment: xx_product=name1
AWS resources: kubernets_xx_product=name1

```
labelMappingConfigs:
          enabled: true
          product_label: xx_product
          product_external_label: kubernetes_xx_product
```
We think this configuration will work for Cost allocation setting **Aggregate by** = Product like it is working for scenario 1. The external costs is still 0 but I could access **Inspect External Costs**. However we also get another **kuberentes_** added to the AWS resource when inspecting external costs. Expected tag/label was **kubernetes_xx_product** but we got **kubernetes_kubernetes_xx_product**.

We are running version 1.92.0.

gz#1746<br><br><i>(related to <a href='https://kubecost.zendesk.com/agent/tickets/1746'>Zendesk ticket #1746</a>)</i>

**Comment by jonerer** [comment]
+1 To this. 

For the view "external costs", it should be noted that using the kubernetes-concept-bound things like "cluster"/"namespace"/"deployment" **works**. In the case of namespace, it adds "kubernetes_" so it's "kubernetes_namespace", which is what the documentation tells us to tag AWS resources with. And it works for both "cost allocation" and "external resources" page.

When it comes to the other "labeled" concepts, "team"/"owner"/"department"/"product"/"environment", it does **not** seem to work for the "external resources" page. It doesn't seem to use the product_external_label as-is. It adds "kubernetes_" in front of it or sometimes changes it completely.

**Comment by kirbsauce** [comment]
Thanks for all the hard work, @serakis!  We'll have a look!

**Comment by kaelanspatel** [comment]
There are definitely issues with the current redirect, thanks for the reports guys.

@serakis posted this in your other issue https://github.com/kubecost/cost-model/issues/1182 but will post here as well for clarity. As detailed below, both the Inspect External Costs functionality and the attached `external.html` will be deprecated in the next version.

> The Inspect External Costs functionality in Cost Allocation will be removed in the upcoming release of `v1.93`, as Cost Allocation will no longer display external costs directly. To view these costs in `v1.93` you can will need to go to the Assets page directly.
>
> Alternatively, you can use the new beta Advanced Reporting feature that will be introduced @ `/advanced-reporting.html` once `v1.93` is released to replicate a similar view. To do so:
>- Set the K8s Allocation Data field to whatever Allocation aggregation you'd like.
>- Set the Cloud Asset Data field to "Service".
> - Set the Data Source Grouping field to whatever label you'd like to use to associate Allocations <-> Assets.
> - You should be able to view a breakdown of the Cloud Assets in a similar way as the current Inspect External Costs.
>
> We encourage you to try this out in the next release. I'm going to close this issue out (again, the Inspect External functionality will be removed in the next version), but if you have any more questions or concerns please don't hesitate to create another issue or reach out to us on Slack!

**Comment by jonerer** [comment]
Thanks for the response @kaelanspatel . But ngl that's a real bummer. Being able to easily correlate external costs with products we run on kubernetes is like 50% of the reason we use kubecost. 

Well check out the advanced reporting feature thanks.  Is there somewhere we can see the future timeline for the kubecost service so we can avoid spending time to support something that's going to be removed in the future? 

Also maybe you should consider a major version bump for that if you follow semver

EDIT:
Oh the advanced reporting feature is not included until v1.93 I see. So external pricing is removed (or just deprecated?) at the same time as the replacement component is introduced.

**Comment by AjayTripathy** [comment]
External pricing correlation is not deprecated @jonerer ! We're just moving it to a new advanced reporting feature in v1.93 at the same time the new component is introduced.
